### ISSUE TITLE
Forced cast to string type

### ISSUE THREAD

**Issue by TimurKhan** [openwisp/django-flat-json-widget#8]
Friends, this widget works with the JSON type. If you do not explicitly specify a quoted string, then you do not need to interpret the value as a string.
Here is an example that actually causes an error in the application due to forced casting of a value to a string type:
`{"max_length": "255"}`

**Comment by nemesisdesign** [comment]
This widget implements what we need in @openwisp, so it's good enough for us, but if you can find a way to retain the right type let us know, I'm open for improvements if they're not too complex.

**Comment by TimurKhan** [comment]
it's not very difficult. You need to override the `format_value `method of the widget. It takes the value of the fields as a string. It needs to be processed. Here is some sample code I sketched for my project. This is the "first pass" (raw code without trying to do it nicely)
```
    def format_value(self, field_value):
        if field_value == '[]':
            return field_value
        data = json.loads(field_value)
        for key, value in data.items():
            if not isinstance(value, str):
                continue
            if value.title in ['True', 'False']:
                try:
                    data[key] = bool(value)
                    continue
                except:
                    pass
            date_time = dateparse.parse_datetime(value)
            if date_time:
                data[key] = dateparse.parse_datetime(value)

                continue

            try:
                data[key] = int(value)
                continue
            except:
                pass

            try:
                data[key] = float(value)
                continue
            except:
                pass

        return json.dumps(data)
```

**Comment by atb00ker** [comment]
Thanks @TimurKhan for sharing your work, if you working on it for your project, could you please create a PR for upstream as well? :smile:

**Comment by TimurKhan** [comment]
Certainly. At a minimum, we always provide links to all code elements that were not written by us, indicating the appropriate licenses and indicating the copyright holder / author.
Well, and on my own behalf, I always recommend tested code to my colleagues. If you have other ideas, I'm ready to do something useful :-)
